File: Java/DailyLeetCode/nAndDouble.java
import java.util.*;
class Solution {
    public boolean checkIfExist(int[] arr) {
        HashMap<Integer,Integer> abc= new HashMap<>();
        
        for(int i=0; i<arr.length; i++){
            if(!abc.containsKey(arr[i])){
                abc.put(arr[i], i);
            }
        }

        for(int i=0; i<arr.length; i++){
            int temp= 2*arr[i];
            if(abc.containsKey(temp)){
                if(abc.get(temp) != i)
                    return true;
            }
        }

        return false;
    }
}

File: Python/DailyLeetCode/nAndDouble.py
from typing import List
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        abc = dict()

        for i in range(len(arr)):
            abc[arr[i]] = i
        
        for i in range(len(arr)):
            temp = 2 * arr[i]
            
            if temp in abc and abc[temp] != i:
                return True
        
        return False
